File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        _map = {}
        i = 0
        indice = 0
        while i < len(nums):
            atual = nums[i]
            complemento = target - nums[i]
            if complemento in _map:
                indice = _map[complemento]
                return [i,indice]
            _map[atual] = i
            i +=1
            


                    
        

        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> vistos = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complemento = target - nums[i];
            if(vistos.containsKey(complemento)){
                return new int[] { vistos.get(complemento), i };
            }
            
            vistos.put(nums[i], i);
        }
        
        return new int[] {}; 
    }
}
